code/game/machinery/Sleeper.dm

```diff
@@ -157,6 +175,40 @@
 /obj/machinery/sleeper/process()
 	..()
 	check_nap_violations()
+	var/mob/living/carbon/C = occupant
+	if(C)
+		if(stasis && C.stat >= stasis_health)
+			C.apply_status_effect(STATUS_EFFECT_STASIS, null, TRUE)
+		else
+			C.remove_status_effect(STATUS_EFFECT_STASIS)
+		switch(active_treatment)
+			if(SLEEPER_TEND)
+				C.heal_bodypart_damage(SLEEPER_HEAL_RATE,SLEEPER_HEAL_RATE) //this is slow as hell, use the rest of medbay you chumps
+			if(SLEEPER_ORGANS)
+				var/heal_reps = efficiency * 2
+				var/list/organs = list(ORGAN_SLOT_EARS,ORGAN_SLOT_EYES,ORGAN_SLOT_LIVER,ORGAN_SLOT_LUNGS,ORGAN_SLOT_STOMACH,ORGAN_SLOT_HEART)
+				for(var/i in 1 to heal_reps)
+					organs = shuffle(organs)
+					for(var/o in organs)
+						var/healed = FALSE
+						var/obj/item/organ/heal_target = C.getorganslot(o)
+						if(heal_target?.damage >= 1)
+							var/organ_healing = 0.2
+							if(C.stat == DEAD)
+								organ_healing = 0.05
+							heal_target.applyOrganDamage(-organ_healing)
+							healed = TRUE
+						if(healed)
+							break
+			if(SLEEPER_CHEMPURGE)
```

Comment:
This needs a nerf, currently it heals 2 tox per tick, and purges between 1 and 4 chems per tick, likely leaning towards the 4. This is significantly better than most if not all purge chems.

Reply:
reduced purging by half, making a fully upgraded sleeper about as effective as pentetic acid